storage_controller/migrations/2025-02-14-160526_safekeeper_timelines/up.sql

```diff
@@ -0,0 +1,19 @@
+CREATE TABLE timelines (
+  tenant_id VARCHAR NOT NULL,
+  timeline_id VARCHAR NOT NULL,
+  start_lsn pg_lsn NOT NULL,
+  generation INTEGER NOT NULL,
+  sk_set BIGINT[] NOT NULL,
+  new_sk_set BIGINT[],
+  cplane_notified_generation INTEGER NOT NULL,
+  deleted_at timestamptz,
+  PRIMARY KEY(tenant_id, timeline_id)
+);
+CREATE TABLE safekeeper_timeline_pending_ops (
+  sk_id BIGINT NOT NULL,
+  tenant_id VARCHAR NOT NULL,
+  timeline_id VARCHAR NOT NULL,
+  generation INTEGER NOT NULL,
+  op_kind VARCHAR NOT NULL,
+  PRIMARY KEY(tenant_id, timeline_id, sk_id)
```

Comment:
@arssher I suppose you want this to be changed to also have the generation? The primary key tuple is supposed to be unique.

Reply:
https://github.com/neondatabase/neon/pull/11058#discussion_r1984856697